File: killrvideo-service-videocatalog/src/main/java/com/killrvideo/service/video/dao/UserVideoRowMapper.java
package com.killrvideo.service.video.dao;

import com.datastax.oss.driver.api.core.cql.Row;
import com.killrvideo.service.video.dto.UserVideo;
import org.springframework.stereotype.Component;

import static com.killrvideo.dse.dto.Video.*;

@Component
public class UserVideoRowMapper {
    public UserVideo map(Row row) {
        return new UserVideo(
                row.getUuid(COLUMN_USERID),
                row.getUuid(COLUMN_VIDEOID),
                row.getString(COLUMN_NAME),
                row.getString(COLUMN_PREVIEW),
                row.getInstant(COLUMN_ADDED_DATE)
        );
    }
}

File: killrvideo-service-videocatalog/src/main/kotlin/com/killrvideo/service/video/dao/UserVideoRowMapper.kt
package com.killrvideo.service.video.dao

import com.datastax.oss.driver.api.core.cql.Row
import com.killrvideo.service.video.dto.UserVideo
import com.killrvideo.service.video.dto.Video
import org.springframework.stereotype.Component

@Component
class UserVideoRowMapper {
    fun map(row: Row): UserVideo =
        UserVideo(
            userid = row.getUuid(Video.COLUMN_USERID),
            addedDate = row.getInstant(Video.COLUMN_ADDED_DATE),
            videoid = row.getUuid(Video.COLUMN_VIDEOID),
            name = row.getString(Video.COLUMN_NAME),
            previewImageLocation = row.getString(Video.COLUMN_PREVIEW)
        )
}
